index.js

```diff
@@ -95,6 +99,15 @@ class Ky {
 			retry: 2,
 			...otherOptions
 		};
+		let abortController;
+
+		if (supportsAbortController) {
+			abortController = new AbortController();
+			if (this._options.signal) {
+				this._options.signal.addEventListener('abort', () => abortController.abort());
```

Comment:
This listener would need to be removed once the request is completed (or errored, i.e. in a `finally`), otherwise the listener would leak and stay around forever even though the request cannot be cancelled anymore

Reply:
Believe it or not, this was never fixed before the PR landed so we ran into this memory leak in our application in 2024! :smile: